### ISSUE TITLE
Invalid JSON for resource embedding error

### ISSUE THREAD

**Issue by begriffs** [begriffs/postgrest#757]
I accidentally tried to embed an incompatible table in another and the JSON of the resulting error message is invalid. Not only is there a message inside a message, but it lacks proper escaping even for that.

```
{"message":"{"details":"no relation between foo and bar","message":"could not find foreign keys between these entities"}"}
```

**Comment by diogob** [comment]
this is probably a regression related to the latest refactoring around the error message. I'll take a look and add some regression testing.

**Comment by begriffs** [comment]
Closing in favor of the more general fix described in #791
